layersvt/device_simulation.cpp

```diff
@@ -103,26 +144,57 @@ std::string GetEnvarValue(const char *name) {
     return value;
 }
 
+#if defined(__ANDROID__)
+void AndroidPrintf(const char *level, const char *fmt, va_list args) {
+    int requiredLength;
+    va_list argcopy;
+    va_copy(argcopy, args);
+    requiredLength = vsnprintf(NULL, 0, fmt, argcopy) + 1;
+    va_end(argcopy);
+
+    char *message = (char *)malloc(requiredLength);
+    vsnprintf(message, requiredLength, fmt, args);
+    __android_log_print(ANDROID_LOG_INFO, "devsim", "%s: %s", level, message);
```

Comment:
This should select the log priority based on level.  There's an example of this in [vktrace_lib.c](https://github.com/LunarG/VulkanTools/blob/master/vktrace/vktrace_layer/vktrace_lib.c#L104)